testing/node_test.go

```diff
@@ -0,0 +1,491 @@
+package testing
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/jackzampolin/horcrux/cmd/horcrux/cmd"
+	"github.com/jackzampolin/horcrux/internal/signer"
+	tmjson "github.com/tendermint/tendermint/libs/json"
+	"github.com/tendermint/tendermint/privval"
+
+	"github.com/avast/retry-go"
+	"github.com/cosmos/cosmos-sdk/client"
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	"github.com/cosmos/cosmos-sdk/simapp/params"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ory/dockertest"
+	"github.com/ory/dockertest/docker"
+	"github.com/stretchr/testify/require"
+	tmconfig "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/p2p"
+	rpcclient "github.com/tendermint/tendermint/rpc/client"
+	rpchttp "github.com/tendermint/tendermint/rpc/client/http"
+	libclient "github.com/tendermint/tendermint/rpc/jsonrpc/client"
+	// "github.com/testcontainers/testcontainers-go/wait"
+)
+
+var (
+	valKey   = "validator"
+	genCoins = "1000000000000stake"
+)
+
+// ChainType represents the type of chain to instantiate
+type ChainType struct {
+	Repository string
+	Version    string
+	Bin        string
+	Ports      map[docker.Port]struct{}
+}
+
+// ChainType instance for simd
+var simdChain = &ChainType{
+	Repository: "jackzampolin/simd",
+	Version:    "v0.42.3",
```

Comment:
Pull this version in dynamically from go.mod https://pkg.go.dev/golang.org/x/mod/modfile